File: src/mysoundtest.py
#!/usr/bin/env python

import rospy
from sound_play.libsoundplay import SoundClient
from std_msgs.msg import Int32


def back():
        sound_client = SoundClient()
        rospy.sleep(2)
        sound = sound_client.waveSound('/home/ubuntu/catkin_ws/src/mysound/soundfile/back.wav')
        sound.play()

def left():
        sound_client = SoundClient()
        rospy.sleep(2)
        sound = sound_client.waveSound('/home/ubuntu/catkin_ws/src/mysound/soundfile/left.wav')
        sound.play()

def leftback():
        sound_client = SoundClient()
        rospy.sleep(2)
        sound = sound_client.waveSound('/home/ubuntu/catkin_ws/src/mysound/soundfile/leftback.wav')
        sound.play()

def right():
        sound_client = SoundClient()
        rospy.sleep(2)
        sound = sound_client.waveSound('/home/ubuntu/catkin_ws/src/mysound/soundfile/right.wav')
        sound.play()

def rightback():
        sound_client = SoundClient()
        rospy.sleep(2)
        sound = sound_client.waveSound('/home/ubuntu/catkin_ws/src/mysound/soundfile/rightback.wav')
        sound.play()

def direction_callback(data):
                #keyboard ja apan cham go
                # (4:left), (1:leftback), (2:back), (6:right), (3:rightback)
                if(data.data == 4):
                        rospy.loginfo("left sound is activated")
                        left()
                elif(data.data == 1):
                        rospy.loginfo("leftback sound is activated")
                        leftback()
                elif(data.data == 2):
                        rospy.loginfo("back sound is activated")
                        back()
                elif(data.data == 6):
                        rospy.loginfo("right sound is activated")
                        right()
                elif(data.data == 3):
                        rospy.loginfo("rightback sound is activated")
                        rightback()

def main():
        rospy.init_node('mysound', anonymous=True)
        sub = rospy.Subscriber('direction', Int32, direction_callback)
        rospy.spin()


if __name__ == '__main__':
        try:
                main()
        except rospy.ROSInterruptException:
                pass

File: src/mysoundtest.cpp
#include "ros/ros.h"
#include "std_msgs/Int32.h"
#include <string>
#include <iostream>
#include <cstdio>
#include <cstring>
#include <memory>
#include <unistd.h>
#include <sound_play/sound_play.h>



void back(){
  sound_play::SoundClient sc;
  sc.playWave("'/home/ubuntu/catkin_ws/src/mysound/soundfile/back.wav'");
  sleep(1);
}

void left(){
  sound_play::SoundClient sc;
  sc.playWave("'/home/ubuntu/catkin_ws/src/mysound/soundfile/left.wav'");
  sleep(1);
}

void leftback(){
  sound_play::SoundClient sc;
  sc.playWave("'/home/ubuntu/catkin_ws/src/mysound/soundfile/leftback.wav'");
  sleep(1);
}

void right(){
  sound_play::SoundClient sc;
  sc.playWave("'/home/ubuntu/catkin_ws/src/mysound/soundfile/right.wav'");
  sleep(1);
}

void rightback(){
  sound_play::SoundClient sc;
  sc.playWave("'/home/ubuntu/catkin_ws/src/mysound/soundfile/rightback.wav'");
  sleep(1);
}


/*//git자료
void playerCallback(const std_msgs::String::ConstPtr& msg) {
  FILE *gp;
  // const std::string filename = "warning1.mp3"; // filename of mp3 file
  std::string message;
  if (msg->data.find(".mp3") != std::string::npos) {
    message = "mpg123 -q " + msg->data;
  } else if (msg->data.find(".wav") != std::string::npos) {
    message = "aplay " + msg->data;
  }else{
    ROS_INFO("%s isn't a sound file.", msg->data.c_str());
  }

  if (!message.empty()) {
    char *cstr = new char[message.size() + 1]; // get memory
    std::strcpy(cstr, message.c_str());        // copy
    gp = popen(cstr, "w");
    pclose(gp); // パイプを閉じる

    ROS_INFO("playing %s", msg->data.c_str());
  }
}
*/



void playerCallback(const std_msgs::Int32::ConstPtr& msg){
  ROS_INFO("Recieved: [%d]", msg->data);
  int waydata = msg->data;
  if (waydata == 4){
    left();
  }
  else if (waydata ==1){
    leftback();
  }
  else if (waydata ==2){
    back();
  }
  else if (waydata ==6){
    right();
  }
  else if (waydata ==3){
    rightback();
  }
}



int main(int argc, char **argv) {

  // establish ros node
  ros::init(argc, argv, "mysound");//노드명 초기화
  ros::NodeHandle nh;//ros통신을 위한 node handle
  sound_play::SoundClient sc;

  ros::Subscriber sub = nh.subscribe("direction", 10, playerCallback);
  ROS_INFO("sound player is ready.");

  ros::spin();

}
